src/pages/ThorChainLP/queries/hooks/useSendThorTx.tsx

```diff
@@ -0,0 +1,442 @@
+// Note, THORChain swaps *are* part of THORFi but there's no notion of enter/exit for them really
+type ThorfiAction = `${ThorfiActionBase}Enter` | `${ThorfiActionBase}Exit` | 'ThorchainSwap'
+
+type Props = {
+  // TODO(gomes): make this a union and pass one or ther other depending on some heuristics?
+  amountCryptoBaseUnit: string | undefined
+  dustAmountCryptoBaseUnit: string | undefined
+  assetId: AssetId
+  // TODO(gomes): we should build this here too, but until this is properly implemented for all we're just passing it
+  // to keep things simple
+  memo: string
+  accountId: AccountId
+  transactionType: 'MsgDeposit' | 'EvmCustomTx' | 'Send'
+  assetAddress: string
+  thorfiAction: ThorfiAction
+}
+
+// TODO(gomes): the hook is for react - extract the non-react parts to smaller bits
+// TODO(gomes): what's a good name for that?
+export const useSendThorTx = ({
+  amountCryptoBaseUnit,
+  dustAmountCryptoBaseUnit,
+  assetId, // i.e the AssetId of the Tx, either native asset, or token to be sent from the contract, assuming allowance has been set
+  memo,
+  accountId,
+  transactionType,
+  assetAddress,
+  thorfiAction,
+}: Props) => {
+  // TODO(gomes): savers sometimes also use dust amounts, ensure this works for them
+  const shouldUseDustAmount = thorfiAction === 'ThorchainLpExit'
+  const [txId, setTxId] = useState<string | null>(null)
+  const [serializedTxIndex, setSerializedTxIndex] = useState<string | null>(null)
+  const wallet = useWallet().state.wallet
+  const toast = useToast()
+  const translate = useTranslate()
+
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const feeAsset = useAppSelector(state =>
+    selectFeeAssetByChainId(state, fromAssetId(assetId).chainId),
+  )
+
+  const accountNumberFilter = useMemo(() => ({ assetId, accountId }), [accountId, assetId])
+  const accountNumber = useAppSelector(s => selectAccountNumberByAccountId(s, accountNumberFilter))
+  const selectedCurrency = useAppSelector(selectSelectedCurrency)
+
+  const amountCryptoPrecision = useMemo(
+    () => fromBaseUnit(amountCryptoBaseUnit ?? 0, asset?.precision ?? 0),
+    [amountCryptoBaseUnit, asset],
+  )
+  const dustAmountCryptoPrecision = useMemo(
+    () => fromBaseUnit(dustAmountCryptoBaseUnit ?? 0, asset?.precision ?? 0),
+    [dustAmountCryptoBaseUnit, asset],
+  )
+
+  const { data: inboundAddressData } = useQuery({
+    ...reactQueries.thornode.inboundAddresses(),
+    staleTime: 60_000,
+    select: data => selectInboundAddressData(data, assetId),
+    enabled: !!assetId,
+  })
+
+  const estimateFeesArgs = useMemo(() => {
+    if (!assetId || !feeAsset || !wallet || !asset) return undefined
+
+    const amountCryptoBaseUnit = toBaseUnit(amountCryptoPrecision, asset.precision)
+
+    // TODO(gomes): ensure that as part of this PR, we do not send dust amount instead of depositWithExpiry() Txs for EVM chains
```

Comment:
Actually not as part of this PR, but keep TODO for when savers is implemented in part 2
```suggestion
    // TODO(gomes): when consuming this in THORCHain savers, ensure we do not send dust amount instead of depositWithExpiry() Txs for EVM chains
```